vm/devices/net/mana_driver/src/gdma_driver.rs

```diff
@@ -241,65 +265,47 @@ struct EqeWaitResult {
 }
 
 impl<T: DeviceBacking> GdmaDriver<T> {
+    /// Unmap (disable) all mapped MSI-X vectors for this driver instance.
+    pub fn unmap_all_msix(&mut self) -> anyhow::Result<()> {
+        let device = match self.device.as_mut() {
+            Some(d) => d,
+            None => return Ok(()),
+        };
+        for (idx, slot) in self.interrupts.iter_mut().enumerate() {
+            if slot.is_some() {
+                if let Err(e) = device.unmap_interrupt(idx as u32) {
```

Comment:
Instead of unmapping one at a time, lets add a function to unmap all interrupts in one go for performance reasons. In vfio.rs, i think you can use `self.interrupts.len()` as the count for the number of interrupts to unmap.